webapp/app/templates/lotse/form_steuernummer.html

```diff
@@ -0,0 +1,184 @@
+{% extends 'basis/form_standard.html' %}
+{% import "macros.html" as macros %}
+
+
+{% block form_content %}
+    <div class="form-row-center">
+        {{ macros.render_field(form.steuernummer_exists, first_field=True if not render_info.form.errors) }}
+    </div>
+    <div class="form-row-center">
+        {{ macros.render_field(form.bundesland) }}
+    </div>
+
+    <div class="form-row-center">
+        {{ macros.render_field(form.bufa_nr) }}
+    </div>
+
+    <div class="form-row-center">
+        {{ macros.render_field(form.steuernummer) }}
+    </div>
+
+    <div class="additional-input-content" for="request_new_tax_number">
+        <h2 class="form-sub-heading-smaller">{{ _('form.lotse.request_new_tax_number.title') }}</h2>
+        {{ _('form.lotse.request_new_tax_number.intro') }}
+        {{ macros.render_field(form.request_new_tax_number) }}
+    </div>
+
+{% endblock %}
+
+{% block optional_js %}
+    {{ super() }}
+    <script src="{{ url_for('static', filename='js/field_utils.js') }}" type=text/javascript></script>
+    <script>
+        let taxOffices = JSON.parse('{{ render_info.additional_info['tax_offices']|tojson }}');
+        let TAX_NUMBER_FORM_STATES = {
+            'NothingSelected': 'NothingSelected',
+            'StateSelectionTaxNumberExists': 'StateSelectionTaxNumberExists',
+            'StateSelectionNoTaxNumberExists': 'StateSelectionNoTaxNumberExists',
+            'TaxOfficeSelection': 'TaxOfficeSelection',
+            'TaxNumberInput': 'TaxNumberInput',
+            'RequestNewTaxNumber': 'RequestNewTaxNumber'
+        }
+        let taxNumberFormState = TAX_NUMBER_FORM_STATES.NothingSelected;
+
+        function updateTaxOfficeList() {
+            let stateDropdown = document.getElementById("bundesland");
+            let bufaNrDropdown = document.getElementById("bufa_nr");
+            var selectedState = stateDropdown.options[stateDropdown.selectedIndex].value.toLowerCase();
+            var selectedTaxOffices = taxOffices.find(taxOffice => taxOffice.state_abbreviation === selectedState)
+            let oldBufaNrValue = bufaNrDropdown.value
+            deleteSelectOptions(bufaNrDropdown, defaultOptionText="{{ _('form.select_input.default_selection') }}");
+
+            if(selectedTaxOffices){
+                selectedTaxOffices = selectedTaxOffices.tax_offices;
+                selectedTaxOffices && selectedTaxOffices.sort(function (a, b) {
+                    return ('' + a.name).localeCompare(b.name);
```

Comment:
The `'' +` probably has a reason here? So a comment could help explain that?

Reply:
It is to make it a String, but I think we could make the conversion explicit and then we would need no comment